File: crimenet_frontend/lib/models/hiring_application.dart
class HiringApplication {
  final int? id;
  final int postId;
  final int applicantId;
  final String coverLetter;
  final DateTime? createdAt;
  final String? status;

  HiringApplication({
    this.id,
    required this.postId,
    required this.applicantId,
    required this.coverLetter,
    this.createdAt,
    this.status,
  });

  factory HiringApplication.fromJson(Map<String, dynamic> json) =>
      HiringApplication(
        id: json['id'],
        postId: json['postId'],
        applicantId: json['applicantId'],
        coverLetter: json['coverLetter'],
        createdAt:
            json['createdAt'] != null
                ? DateTime.parse(json['createdAt'])
                : null,
        status: json['status'],
      );

  Map<String, dynamic> toJson() => {
    if (id != null) 'id': id,
    'postId': postId,
    'applicantId': applicantId,
    'coverLetter': coverLetter,
  };
}

File: app/src/main/java/org/example/entity/HiringApplication.java
package org.example.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class HiringApplication {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    private HiringPost post;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "applicant_id")
    private User applicant;

    private String coverLetter;
    private LocalDateTime createdAt;

    @Enumerated(EnumType.STRING)
    private ApplicationStatus status;

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public HiringPost getPost() { return post; }
    public void setPost(HiringPost post) { this.post = post; }
    public User getApplicant() { return applicant; }
    public void setApplicant(User applicant) { this.applicant = applicant; }
    public String getCoverLetter() { return coverLetter; }
    public void setCoverLetter(String coverLetter) { this.coverLetter = coverLetter; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public ApplicationStatus getStatus() { return status; }
    public void setStatus(ApplicationStatus status) { this.status = status; }
} 